### DISCUSSION TITLE
With BottomBar

### DISCUSSION THREAD

**Discussion by mrakramov** [bumble-tech/appyx#106] [Q&A] [answered]
Is it possible to use the lib with bottom bar?

**Comment by KovalevAndrey** [comment, marked as answer]
Yes, absolutely. `BottomNavigation` is just a composable function that has children as `BottomNavigationItem` composable. You define how to handle onClick events. In this case you will need to add `BottomNavigation` to the ParentNode view and perform routing operations like `backstack.push(..)` or any others in order to change routings when user clicks on the BottomNavigationItem.

**Reply by mrakramov** [reply]
I have tested it and it works well, But there is one problem here. How can I set the selected bottom item?

**Reply by KovalevAndrey** [reply]
@mrakramov you just need to set BottomNavigationItem as selected
```@Composable
fun RowScope.BottomNavigationItem(
    selected: Boolean
```

to understand which routing is currently shown in `BackStack` there's an API for that: 
`com.bumble.appyx.navmodel.backstack.activeRouting`

**Reply by mrakramov** [reply]
![image](https://user-images.githubusercontent.com/51120212/186893802-224a5fb1-3c13-4b10-a7d9-6929d5b4db03.png)
it is what i did, but when i select the 2nd item, screen changes but bottomSelected doesn't change

**Reply by KovalevAndrey** [reply]
Because `backStack.activeRouting` is not a compose State object your composable function will not update when active routing is changed. You need a State object something like this
```
val currentlyActiveChild = remember(backStack.activeRouting) {
            mutableStateOf(backStack.activeRouting)
        }
```
```
selected = currentlyActiveChild.value == screen.route
```
also you may want consider using `com.bumble.appyx.navmodel.spotlight.Spotlight` instead of `BackStack` if you need to change tabs without re-creating new items every time

**Reply by mrakramov** [reply]
thnx a lot. I do it with Spotlight. It works perfect now

**Reply by mrakramov** [reply]
val currentlyActiveChild = remember(backStack.activeRouting) {
            mutableStateOf(backStack.activeRouting)
}
selected = currentlyActiveChild.value == screen.route
I have tested it out, but It doesn't work.

**Reply by KovalevAndrey** [reply]
looks like for some reason recomposition is not triggered for this `BottomNavigationItem` when active element is changed, try to 
1) verify that state is changed when a new element is pushed to backstack
2) that your condition is true

**Reply by mrakramov** [reply]
ok, thnx))

**Reply by Wysel** [reply]
Hi @KovalevAndrey , how would you recommend using Scaffold with top/bottom bar? Thanks.

**Reply by Wysel** [reply]
I just realized that Children(...) is just regular composable and placing it within a content of Scaffold will do it.

**Reply by Jparrgam** [reply]
hi @KovalevAndrey I followed your advice to use Spotlight but I have a problem when I change tabs with this method

spotlight.previous()

the composable is completely recreated

**Reply by KovalevAndrey** [reply]
hi @Jparrgam, this is expected behaviour. When you switch between children Node by executing `next` and `previous` operations, UI of the Nodes leaves the composition when transition is finished. The state of UI is saved and will be restored when it enters the composition once again
